### ISSUE TITLE
Error: must be called with a dataclass type or instance

### ISSUE THREAD

**Issue by robotzheng** [sii-research/siiRL#5]
partitioned_dataset.py:267:_load_partitioned_raw_data >> Failed during partitioned data loading for DDP rank 0: ['/mnt/workspace/group/zzt/RL_test/siiRL/data/datasets/geo3k/train.parquet']. Error: must be called with a dataclass type or instance

**Comment by robotzheng** [comment]
absl-py                                  2.1.0
accelerate                               1.9.0
addict                                   2.4.0
aiofiles                                 23.2.1
aiohappyeyeballs                         2.4.4
aiohttp                                  3.11.10
aiohttp-cors                             0.8.1
aiosignal                                1.3.1
airportsdata                             20250706
alibabacloud-oss-v2                      1.1.2
aliyun-python-sdk-core                   2.16.0
aliyun-python-sdk-kms                    2.16.5
altair                                   5.5.0
annotated-types                          0.7.0
antlr4-python3-runtime                   4.9.3
anyio                                    4.7.0
astor                                    0.8.1
asttokens                                3.0.0
async-timeout                            5.0.1
attrs                                    24.2.0
audioread                                3.0.1
audioseal                                0.1.4
auraloss                                 0.4.0
av                                       15.0.0
beautifulsoup4                           4.12.3
blake3                                   1.0.5
cachetools                               5.5.0
certifi                                  2024.8.30
cffi                                     1.17.1
charset-normalizer                       3.4.0
click                                    8.1.7
cloudpickle                              3.1.1
codetiming                               1.4.0
coloredlogs                              15.0.1
colorful                                 0.5.7
comm                                     0.2.3
compressed-tensors                       0.9.3
conformer                                0.3.2
contourpy                                1.3.1
crcmod                                   1.7
crcmod-plus                              2.1.0
cryptography                             44.0.0
cupy-cuda12x                             13.5.1
curlify                                  3.0.0
cycler                                   0.12.1
Cython                                   3.0.11
dacite                                   1.9.2
datasets                                 2.18.0
debugpy                                  1.8.15
decorator                                5.1.1
deepspeed                                0.14.2
Deprecated                               1.2.18
depyf                                    0.18.0
diffusers                                0.29.0
dill                                     0.3.8
diskcache                                5.6.3
distlib                                  0.4.0
distro                                   1.9.0
dnspython                                2.7.0
editdistance                             0.8.1
einops                                   0.8.0
einx                                     0.3.0
email_validator                          2.2.0
exceptiongroup                           1.2.2
executing                                2.2.0
fastapi                                  0.115.6
fastapi-cli                              0.0.8
fastapi-cloud-cli                        0.1.5
fastrlock                                0.8.3
ffmpy                                    0.4.0
filelock                                 3.16.1
flatbuffers                              24.3.25
fonttools                                4.55.3
frozendict                               2.4.6
frozenlist                               1.5.0
fsspec                                   2024.2.0
funasr                                   1.2.0
gast                                     0.6.0
gdown                                    5.1.0
gguf                                     0.17.1
gitdb                                    4.0.12
GitPython                                3.1.45
google-api-core                          2.25.1
google-auth                              2.37.0
google-auth-oauthlib                     1.0.0
googleapis-common-protos                 1.70.0
gradio                                   5.4.0
gradio_client                            1.4.2
grpcio                                   1.57.0
h11                                      0.14.0
hf-xet                                   1.1.5
hjson                                    3.1.0
httpcore                                 1.0.7
httptools                                0.6.4
httpx                                    0.28.1
huggingface-hub                          0.34.3
humanfriendly                            10.0
hydra-core                               1.3.2
HyperPyYAML                              1.2.2
idna                                     3.10
imageio                                  2.37.0
importlib_metadata                       8.0.0
importlib_resources                      6.4.5
inflect                                  7.3.1
interegular                              0.3.3
ipykernel                                6.30.0
ipython                                  8.37.0
jaconv                                   0.4.0
jamo                                     0.4.1
jedi                                     0.19.2
jieba                                    0.42.1
Jinja2                                   3.1.4
jiter                                    0.10.0
jmespath                                 0.10.0
joblib                                   1.4.2
jsonschema                               4.23.0
jsonschema-specifications                2024.10.1
julius                                   0.2.7
jupyter_client                           8.6.3
jupyter_core                             5.8.1
kaldiio                                  2.18.0
kiwisolver                               1.4.7
lark                                     1.2.2
latex2sympy2_extended                    1.10.2
lazy_loader                              0.4
librosa                                  0.10.2
lightning                                2.2.4
lightning-utilities                      0.11.9
llguidance                               0.7.30
llvmlite                                 0.44.0
lm-format-enforcer                       0.10.11
loguru                                   0.7.3
Markdown                                 3.7
markdown-it-py                           3.0.0
MarkupSafe                               2.1.5
math-verify                              0.8.0
mathruler                                0.1.0
matplotlib                               3.7.5
matplotlib-inline                        0.1.7
mdurl                                    0.1.2
mistral_common                           1.8.3
modelscope                               1.15.0
more-itertools                           10.5.0
mpmath                                   1.3.0
msgpack                                  1.1.0
msgspec                                  0.19.0
multidict                                6.1.0
multiprocess                             0.70.16
narwhals                                 1.17.0
nest-asyncio                             1.6.0
networkx                                 3.1
ninja                                    1.11.1.2
nnAudio                                  0.3.3
numba                                    0.61.2
numpy                                    2.2.6
nvidia-cublas-cu12                       12.4.5.8
nvidia-cuda-cupti-cu12                   12.4.127
nvidia-cuda-nvrtc-cu12                   12.4.127
nvidia-cuda-runtime-cu12                 12.4.127
nvidia-cudnn-cu12                        9.1.0.70
nvidia-cufft-cu12                        11.2.1.3
nvidia-curand-cu12                       10.3.5.147
nvidia-cusolver-cu12                     11.6.1.9
nvidia-cusparse-cu12                     12.3.1.170
nvidia-cusparselt-cu12                   0.6.2
nvidia-ml-py                             12.560.30
nvidia-nccl-cu12                         2.21.5
nvidia-nvjitlink-cu12                    12.4.127
nvidia-nvtx-cu12                         12.4.127
oauthlib                                 3.2.2
omegaconf                                2.3.0
onnx                                     1.16.0
onnxconverter-common                     1.14.0
onnxmltools                              1.13.0
onnxruntime-gpu                          1.16.0
openai                                   1.95.1
openai-whisper                           20231117
opencensus                               0.11.4
opencensus-context                       0.1.3
opencv-python-headless                   4.12.0.88
opentelemetry-api                        1.26.0
opentelemetry-exporter-otlp              1.26.0
opentelemetry-exporter-otlp-proto-common 1.26.0
opentelemetry-exporter-otlp-proto-grpc   1.26.0
opentelemetry-exporter-otlp-proto-http   1.26.0
opentelemetry-exporter-prometheus        0.47b0
opentelemetry-proto                      1.26.0
opentelemetry-sdk                        1.26.0
opentelemetry-semantic-conventions       0.47b0
opentelemetry-semantic-conventions-ai    0.4.12
orjson                                   3.10.12
oss2                                     2.19.1
outlines                                 0.1.11
outlines_core                            0.1.26
packaging                                25.0
pandas                                   2.2.3
parso                                    0.8.4
partial-json-parser                      0.2.1.1.post6
peft                                     0.16.0
pexpect                                  4.9.0
pillow                                   10.4.0
pip                                      25.1
platformdirs                             4.3.6
pooch                                    1.8.2
prometheus_client                        0.22.1
prometheus-fastapi-instrumentator        7.1.0
prompt_toolkit                           3.0.51
propcache                                0.2.1
proto-plus                               1.26.1
protobuf                                 3.20.2
psutil                                   6.1.0
ptyprocess                               0.7.0
pure_eval                                0.2.3
py-cpuinfo                               9.0.0
py-spy                                   0.4.0
pyarrow                                  21.0.0
pyarrow-hotfix                           0.7
pyasn1                                   0.6.1
pyasn1_modules                           0.4.1
pybind11                                 3.0.0
pycountry                                24.6.1
pycparser                                2.22
pycryptodome                             3.21.0
pydantic                                 2.11.7
pydantic_core                            2.33.2
pydantic-extra-types                     2.10.5
pydub                                    0.25.1
Pygments                                 2.18.0
pylatexenc                               2.10
pynini                                   2.1.5
pynndescent                              0.5.13
pynvml                                   12.0.0
pyparsing                                3.2.0
PySocks                                  1.7.1
python-dateutil                          2.9.0.post0
python-dotenv                            1.0.1
python-json-logger                       3.3.0
python-multipart                         0.0.12
pytorch-lightning                        2.4.0
pytorch-wpe                              0.0.1
pytz                                     2024.2
pyvers                                   0.1.0
pyworld                                  0.3.4
PyYAML                                   6.0.2
pyzmq                                    27.0.0
qwen-vl-utils                            0.0.11
ray                                      2.47.1
referencing                              0.35.1
regex                                    2024.11.6
requests                                 2.32.3
requests-oauthlib                        2.0.0
rich                                     13.7.1
rich-toolkit                             0.14.9
rignore                                  0.6.4
rpds-py                                  0.22.3
rsa                                      4.9
ruamel.yaml                              0.18.6
ruamel.yaml.clib                         0.2.12
ruff                                     0.8.2
safehttpx                                0.1.6
safetensors                              0.4.5
scikit-learn                             1.6.0
scipy                                    1.14.1
semantic-version                         2.10.0
sentencepiece                            0.2.0
sentry-sdk                               2.34.1
setuptools                               78.1.1
shellingham                              1.5.4
siirl                                    0.0.2         /mnt/workspace/group/zzt/RL_test/siiRL
simplejson                               3.19.3
six                                      1.17.0
smart_open                               7.3.0.post1
smmap                                    5.0.2
sniffio                                  1.3.1
sortedcontainers                         2.4.0
soundfile                                0.12.1
soupsieve                                2.6
soxr                                     0.5.0.post1
sseclient-py                             1.8.0
stack-data                               0.6.3
starlette                                0.41.3
sympy                                    1.13.1
tensorboard                              2.14.0
tensorboard-data-server                  0.7.2
tensorboardX                             2.6.2.2
tensordict                               0.9.1
tensorrt-cu12                            10.0.1
tensorrt-cu12-bindings                   10.0.1
tensorrt-cu12-libs                       10.0.1
threadpoolctl                            3.5.0
tiktoken                                 0.8.0
timm                                     1.0.19
tokenizers                               0.21.4
tomli                                    2.2.1
tomlkit                                  0.12.0
torch                                    2.6.0
torch-complex                            0.4.4
torchaudio                               2.6.0
torchdata                                0.11.0
torchmetrics                             1.6.0
torchvision                              0.21.0
tornado                                  6.5.1
tqdm                                     4.67.1
traitlets                                5.14.3
transformers                             4.54.1
triton                                   3.1.0
ttsfrd                                   0.4.2
ttsfrd-dependency                        0.1
typeguard                                4.4.1
typer                                    0.15.1
typing_extensions                        4.12.2
typing-inspection                        0.4.1
tzdata                                   2024.2
ujson                                    5.10.0
umap-learn                               0.5.7
urllib3                                  2.2.3
uvicorn                                  0.30.0
uvloop                                   0.21.0
virtualenv                               20.32.0
vllm                                     0.8.5.post1
wandb                                    0.21.0
watchfiles                               1.0.3
wcwidth                                  0.2.13
websockets                               11.0.3
Werkzeug                                 3.1.3
WeTextProcessing                         1.0.3
wget                                     3.2
wheel                                    0.45.1
wrapt                                    1.17.2
x-transformers                           2.3.5
xformers                                 0.0.29.post2
xgrammar                                 0.1.18
xxhash                                   3.5.0
yapf                                     0.43.0
yarl                                     1.18.3
zipp                                     3.21.0

**Comment by liao1995** [comment]
Hi,

Thanks for providing your environment details.

We've compared our environments and noticed that your datasets version is 2.18.0, while we are using 3.5.0. We attempted to reproduce the issue by downgrading our datasets version to 2.18.0, but we were unable to replicate the error on our end.

This suggests a couple of potential paths forward. Could you please try one of the following?

Upgrade your datasets library to version 3.5.0 or higher (pip install --upgrade datasets) and see if the issue persists.

Provide the full error stack trace. If upgrading doesn't solve it, the complete error output, showing which specific line of code is failing, would give us the information we need for further analysis.

We appreciate your cooperation as we work to resolve this.

**Comment by liao1995** [comment]
Hi,

Just wanted to follow up on this issue. Were you able to resolve the error by upgrading the datasets library?

Please let us know if you're still experiencing problems or have any other questions. If we don't hear back, we'll assume the issue is resolved and close this ticket in a few days.

Thanks!

**Comment by robotzheng** [comment]
it's OK. The project is running well.

**Comment by liao1995** [comment]
That's great to hear it! Thanks for confirming.

I'll go ahead and close this issue now. Feel free to reopen it or create a new one if you run into any other problems.

Cheers!
